---
title: "Assignment_3"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

###Problem 1####
The PFAM domain distribution for human proteome can be found at ftp://ftp.ebi.ac.uk/pub/databases/Pfam/current_release/proteomes/9606.tsv.gz. The first column of this file is the protein accession number. The location of the domain hit for each gene is given by the columns 2-5. Columns 2-3 are alignment start and end. Columns 4-5 are envelope start and end. Envelopes are generally considered the location of a domain on a gene. Write a R scrpt that takes 9606.tsv.gz file as a first argument, a protein accession number as a second argument, and a location (integer) as a third argument. The program should print the domain name (hmm_name), if the location falls within a domain for a given protein accession. The program should return nothing if the position is outside the boundaries of domains. We should be able to run the program like this

> problem1.R ../data/9606.tsv.gz O95931 20
> Chromo

Hint: You should create a list using the protein accession as key and location start and end as values. You
might want to create a nested list or two separate lists.


```{bash}
curl -O ftp://ftp.ebi.ac.uk/pub/databases/Pfam/current_release/proteomes/9606.tsv.gz

#gunzip 9606.tsv.gz
```
```{r}
args <- commandArgs(trailingOnly = T)
file <- args[1] #"9606.tsv.gz"
accnum <- args[2] #"A0A024QZ18"
location <- as.integer(args[3]) #"75"

#I changed the example to one that would work


#gunzip and read in the file
data_table = read.table(gzfile(paste0("./9606.tsv.gz")), header=F)
data_table = read.table(gzfile(paste(file)), header=F)

#filter for only the accnum of interest
by_accnum = data_table[which(data_table[ ,1] == "A0A024QZ18"), ]
by_accnum = data_table[which(data_table[,1]==accnum),]


#extract and save envelope start and end
env_start=as.numeric(by_accnum[4])
env_end=as.numeric(by_accnum[5])


#if else statement to see if location is within envelope
#if(75>=env_start && 75<=env_end){
#  cat(paste0(as.matrix(by_accnum[7]),"\n"))
#}else{
#  break()
#}

if(location >= env_start && location <= env_end){
  cat(paste(as.matrix(by_accnum[7]),"\n"))
}else{
  break()
}

####script saved as problem3_1.R
```

```{bash}

#Run R scrip in bash
Rscript problem3_1.R "./9606.tsv.gz" A0A024QZ18 75
```

Swissvar is a database of human gene, their variations, and disease associations. The file can be downloaded
from here: ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/variants/hu
msavar.txt. The 2nd column of the file is the protein accession numbers. These are the same accession
numbers used in the domain file in Problem 1. The 6th column is dbSNP and reports the variation at a
particular location. Using these two files, create a sorted list of domains according to the total number of
their variations. The domains with higher variations should be on top. The program should not take any
argument and output the domain list on STDOUT. The output should have two columns, separated by tab:
domain name (hmm_name) and a number indicating variation, like this:
Domain Variation
BRAC1 150
Chromo 100
...
Remember, your output will differ from the above shown output. The first line is the header. Note: You
may skip writing a run.sh file for this problem.
Hint:
1. If needed, the location of the variation can be extracted from the "AA change" column of SwissVar
data. For e.g., p.His52Arg means the variation is at the location 52.
2. Parsing â€˜humsavar.txt" file is tricky. You may use the following R code to get a clean data.


















